controllers/space/nstemplatetier_spaces_mapper_test.go

```diff
@@ -0,0 +1,103 @@
+package space_test
+
+import (
+	"testing"
+
+	tierutil "github.com/codeready-toolchain/host-operator/controllers/nstemplatetier/util"
+	"github.com/codeready-toolchain/host-operator/controllers/space"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	tiertest "github.com/codeready-toolchain/host-operator/test/nstemplatetier"
+	spacetest "github.com/codeready-toolchain/host-operator/test/space"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestMapNSTemplateTierToSpaces(t *testing.T) {
+
+	// given
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+	err := apis.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+
+	t.Run("single match", func(t *testing.T) {
+		// given
+		nsTmplTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+		outdatedSpace := spacetest.NewSpace("oddity",
+			spacetest.WithTierName(nsTmplTier.Name),
+			spacetest.WithLabel(tierutil.TemplateTierHashLabelKey(nsTmplTier.Name), "outdated"), // label must exist, but with an outdated value compared to the current NSTemplateTier
+		)
+		hostClient := test.NewFakeClient(t, nsTmplTier, outdatedSpace)
```

Comment:
that was already done in https://github.com/codeready-toolchain/host-operator/pull/597/files#diff-d2b7a2e1a0af3303f1ab5a2128fc30afac822018ab7fd35098632d2cb1bf546cR66-R68 but since you're asking for it, then I will add them to all other tests

Reply:
but this uses the same tier, I was thinking about spaces for different tiers

Reply:
I mean, this
```
uptodateSpace := spacetest.NewSpace("oddity3",
    spacetest.WithTierNameAndHashLabelFor(nsTmplTier),
)
```
is totally correct. We could add just additional noise with spaces using totally different tier.

Reply:
see 5b4ab3a